src/components/text.js

```diff
@@ -54,8 +50,14 @@ export default class Text extends Component {
     window.addEventListener('load', this.resize);
     window.addEventListener('resize', this.resize);
   }
-  componentWillReceiveProps() {
-    this.resize();
+  static getDerivedStateFromProps(nextProps, prevState) {
+    return nextProps.fit !== prevState.fit ? { fit: nextProps.fit } : null;
+  }
+
+  componentDidUpdate(prevState, prevProps) {
```

Comment:
This function signature is incorrect – `componentDidUpdate` takes `prevProps` as the first argument and `prevState` as an optional second argument. Unfortunately, this means that the `prevProps` reference below is actually to `prevState` 😱.